src/shadowbox/server/manager_service.ts

```diff
@@ -228,7 +229,18 @@ export class ShadowsocksManagerService {
   public createNewAccessKey(req: RequestType, res: ResponseType, next: restify.Next): void {
     try {
       logging.debug(`createNewAccessKey request ${JSON.stringify(req.params)}`);
-      this.accessKeys.createNewAccessKey().then((accessKey) => {
+      let encryptionMethod = req.params.method;
+      logging.debug(`encryptionMethod: $(encryptionMethod)`);
+      if (!encryptionMethod) {
+        encryptionMethod = '';
+      }
+      if (typeof encryptionMethod !== 'string') {
+        return next(new restifyErrors.InvalidArgumentError(
+            {statusCode: 400},
+            `Expected a string encryptionMethod, instead got ${encryptionMethod} of type ${
+                typeof encryptionMethod}`));
+      }
+      this.accessKeys.createNewAccessKey(encryptionMethod).then((accessKey) => {
```

Comment:
You should capture the invalid cipher error in the catch below. See `setPortForNewAccessKeys` for an example. You you can use the invalid cipher error that you defined and didn't use.

Reply:
done.